### PR TITLE
evaluator,parser: support ASCII()

### PR DESCRIPTION
**Pull request by zyguan**
Hi, I'm reading the source code and willing to help to implement some basic builtin functions. However, I'm not sure if I did things right. Please correct me if I made mistake.

### CODE REVIEW

**Comment by shenli** [comment]
@zyguan Thanks  for your PR. Please sign the CLA.

File: parser/parser.y [hunk 1]
```diff
@@ -2176,13 +2177,17 @@ Function:
 |	FunctionCallAgg
 
 FunctionNameConflict:
-	"DATABASE" | "SCHEMA" | "IF" | "LEFT" | "REPEAT" | "CURRENT_USER" | "CURRENT_DATE" | "VERSION" | "UTC_DATE"
+	"ASCII" | "DATABASE" | "SCHEMA" | "IF" | "LEFT" | "REPEAT" | "CURRENT_USER" | "CURRENT_DATE" | "VERSION" | "UTC_DATE"
```
Comment (shenli) [inline parser/parser.y]: ASCII is not a reserved keyword. It should not be here. You can put it into UnReservedKeyword rule: https://github.com/pingcap/tidb/blob/master/parser/parser.y#L1819

See: https://dev.mysql.com/doc/refman/5.7/en/keywords.html

**Comment by zyguan** [comment]
Ok, I put it there because of [sql_yacc.yy](https://github.com/twitter/mysql/blob/master/sql/sql_yacc.yy). I'll fix it later.

**Comment by zyguan** [comment]
@shenli I see ```COALESCE``` is a nonreserved keyword according to https://dev.mysql.com/doc/refman/5.7/en/keywords.html. However, it appears in ```NotKeywordToken``` in  https://github.com/pingcap/tidb/blob/master/parser/parser.y. I'm a little confused.

**Comment by shenli** [comment]
@zyguan I think it is a mistake. Good catch!

**Comment by shenli** [comment]
LGTM
@coocood PTAL

**Comment by coocood** [comment]
LGTM
Thank you!

**Comment by shenli** [comment]
@zyguan Thanks for your contribution! May I have your email address?

**Comment by zyguan** [comment]
@shenli Sure! zhongyangguan@gmail.com

**Comment by ngaut** [comment]
Thank you, This branch is out-of-date, Could you merge or rebase master ? @zyguan

**Comment by zyguan** [comment]
@ngaut Yes, Of course! Wait a minute.

**Comment by ngaut** [comment]
Thanks @zyguan

**Pull request merged**